python/common/python/time.java

```diff
@@ -106,7 +106,7 @@ public time() {
         return new org.python.types.Str(ft.format(currentDate));
     }
     public static org.python.Object ctime(org.python.Object seconds) {
-        java.util.Date currentDate = new java.util.Date(((org.python.types.Int) seconds.__int__()).value * 1000);
+        java.util.Date currentDate = new java.util.Date((long) (((org.python.types.Int) seconds.__int__()).value * 1000));
```

Comment:
Would it be a good idea to cast to long *before* multiplying?

Reply:
I am not sure if `int` would be large enough hence, I used `(long)` as the seconds could be a large.

Reply:
I understand the line of thinking, but it’s the implementation I’m having doubts about. With the current implementation, the number is first times 1000, and then casted to long. This wouldn’t prevent anything because the cast happens after the overflow (if there is one).

To actually prevent the overflow you’d need to do something like

```java
new java.util.Date(((long) ((org.python.types.Int) seconds.__int__()).value) * 1000)
```

or (probably better)

```java
new java.util.Date(((org.python.types.Int) seconds.__int__()).value * 1000L)
```

so the large number is casted to long first, and then multiplied.

Reply:
Thanks.